lib/schema/schema.js

```diff
@@ -24,8 +24,14 @@ var util = require('util');
 
 var misc = require('../misc.js');
 
+exports.validDeviceTime = function(val) {
+  if (!/^(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d)$/.test(val)) {
```

Comment:
Yeah, then I would probably do a first-pass regex validation just as:

```javascript
    /^(\d{4}-\d\d-\d\dT\d\d:\d\d:\d\d)$/
```

And then follow that up with the parsing.